nautobot/ipam/querysets.py

```diff
@@ -184,6 +183,43 @@ def get_queryset(self):
         """
         return super().order_by("host")
 
+    def string_search(self, search):
+        """
+        Interpret a search string and return useful results.
+        """
+        if not search:
+            return self.none()
+
+        network = self._parse_as_network_string(search)
+        broadcast = self._get_broadcast(network)
+        return self.filter(
+            Q(dns_name__icontains=search)
+            | Q(description__icontains=search)
+            | Q(host__lte=broadcast, host__gte=network.network)  # same as `net_host_contained()`
+        )
+
+    def _parse_as_network_string(self, search):
+        """
+        Attempts to parse a (potentially incomplete) IPAddress and return an IPNetwork.
+        eg: '10.10' should be interpreted as netaddr.IPNetwork('10.10.0.0/16')
```

Comment:
Don't we need similar logic for IPv6 addresses? Will almost certainly be more complicated due to the myriad of ways to enter a full or partial v6 address -  "2001:db8:", "2001:0db8::", "2001:DB8::1", etc.